### ISSUE TITLE
Lint/ConstantDefinitionInBlock detects class definitions inside blocks

### ISSUE THREAD

**Issue by tas50** [rubocop-hq/rubocop#8709]
## Expected behavior

Nothing should be detected

## Actual behavior

A class defined within a block is detected as a constant

## Steps to reproduce the problem

Run `Lint/ConstantDefinitionInBlock` against:

```ruby
describe Ohai::Mixin::ShellOut, "shell_out" do
  class DummyPlugin
    include Ohai::Mixin::ShellOut
  end
end
```

```
/Users/tsmith/test.rb:2:3: W: Lint/ConstantDefinitionInBlock: Do not define constants within a block.
  class DummyPlugin ...
  ^^^^^^^^^^^^^^^^^
```

## RuboCop version

master

**Comment by tas50** [comment]
Similarly this is being detected as a constant within an rspec describe block:

  before do
    PasswdEntry = Struct.new(:name, :uid, :gid, :dir, :shell, :gecos)
  end

**Comment by eugeneius** [comment]
Those are legitimate offenses: both constants end up defined in the global scope, and can be referenced anywhere else in the program - you could write production code that depends on them, for example.
